<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>RNA-seq analysis with Bioconductor: Mapping reads to a reference genome</title><meta name="viewport" content="width=device-width, initial-scale=1"><script src="assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="assets/styles.css"><script src="assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="manifest" href="site.webmanifest"><link rel="mask-icon" href="safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" media="(prefers-color-scheme: light)" content="white"><meta name="theme-color" media="(prefers-color-scheme: dark)" content="black"></head><body>
    <header id="top" class="navbar navbar-expand-md top-nav rcac"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="rcac-logo" alt="rcac" src="assets/images/rcac-logo.svg"></div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text"><li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul></li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Learner View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='instructor/01b-mapping-reads.html';">Instructor View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav rcac" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="rcac" src="assets/images/rcac-logo-sm.svg"></div>
    <div class="lesson-title-md">
      RNA-seq analysis with Bioconductor
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            RNA-seq analysis with Bioconductor
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="reference.html#glossary">Glossary</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="profiles.html">Learner Profiles</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"></ul></li>
      </ul></div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  RNA-seq analysis with Bioconductor
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 16%" class="percentage">
    16%
  </div>
  <div class="progress rcac">
    <div class="progress-bar rcac" role="progressbar" style="width: 16%" aria-valuenow="16" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text"><li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul></li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Learner View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="instructor/01b-mapping-reads.html">Instructor View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->

            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Setup</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-intro-to-rnaseq.html">1. Introduction to RNA-seq</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="01a-from-reads-to-counts.html">2. Downloading and organizing files</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        3. Mapping reads to a reference genome
        </span>
      </button>
    </div><!--/div.accordion-header-->

    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#introduction">Introduction</a></li>
<li><a href="#preparing-for-read-mapping">Preparing for read mapping</a></li>
<li><a href="#indexing">Indexing</a></li>
<li><a href="#mapping-reads-to-the-reference-genome">Mapping reads to the reference genome</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="01c-count-features.html">4. Quantifying Gene Expression</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="02-setup.html">5. RStudio Project and Experimental Data</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="03-import-annotate.html">6. Importing and annotating quantified data into R</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="04-exploratory-qc.html">7. Exploratory analysis and quality control</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="05-differential-expression.html">8. Differential expression analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="06-extra-design.html">9. Extra exploration of design matrices</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="07-gene-set-analysis.html">10. Gene set enrichment analysis</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="08-next-steps.html">11. Next steps</a>
    </div><!--/div.accordion-header-->

  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="reference.html#glossary">Glossary</a>
                      </li>
                      <li>
                        <a href="profiles.html">Learner Profiles</a>
                      </li>

                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources"><a href="aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">

            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="01a-from-reads-to-counts.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="01c-count-features.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="01a-from-reads-to-counts.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Downloading and
        </a>
        <a class="chapter-link float-end" href="01c-count-features.html" rel="next">
          Next: Quantifying Gene...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Mapping reads to a reference genome</h1>
        <p>Last updated on 2024-09-10 |

        <a href="https://aseetharam.github.io/rcac_rnaseq_workshop/edit/main/episodes/01b-mapping-reads.Rmd" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>



        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can we map raw reads to a reference genome?</li>
<li>What programs are available for mapping reads to a reference
genome?</li>
<li>What opitons and parameters need to be considered when mapping?</li>
<li>How do to access the mapping results?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Understand the process of mapping raw reads to a reference
genome.</li>
<li>Fine tune the mapping process by adjusting parameters.</li>
<li>Access the mapping results.</li>
</ul></div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h2>
<hr class="half-width"><p>In this episode, we will learn how to map raw reads to a reference
genome. Mapping is the process of aligning short reads to a reference
genome. The goal of mapping is to determine the location of the reads in
the reference genome. We will use the RNA-seq data that we downloaded
(soft-linked) in the previous episode and map them to the reference
genome using the STAR aligner. STAR is a fast and accurate RNA-seq
aligner that can align reads to the reference genome with high
sensitivity and specificity.</p>
<div id="what-mapping-programs-should-i-use" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="what-mapping-programs-should-i-use" class="callout-inner">
<h3 class="callout-title">What mapping programs should I use?</h3>
<div class="callout-content">
<p>One major requirement for a suitable RNAseq mapping program is that
it should be splice-aware. This means the program must be able to handle
reads that span exon-exon junctions, which is crucial for RNAseq data
since the reads are derived from spliced transcripts. <code>STAR</code>
is a widely used splice-aware aligner for RNAseq, known for its speed
and accuracy when mapping reads to a reference genome. Other popular
RNAseq aligners include <code>HISAT2</code> and, for certain
applications, <code>Bowtie2</code>. It’s important not to use DNA
aligners like <code>BWA</code>, which are not designed to handle spliced
reads.</p>
</div>
</div>
</div>
<div id="how-does-read-quality-affect-alignment-performance" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="how-does-read-quality-affect-alignment-performance" class="callout-inner">
<h3 class="callout-title">How does read quality affect alignment performance?</h3>
<div class="callout-content">
<p>The quality of the reads can have a significant impact on the
alignment performance and thus your downstream analysis.</p>
<ul><li>
<strong>Uneven read quality</strong>: poor quality of the reads can
result in partial alignments or truncated read mapping, leaving you with
large portions of the read unmapped.</li>
<li>
<strong>Overrepresented sequences</strong>: such as ribosomal RNA,
or microbial contamination, can lead to too many multi-mapped
reads.</li>
<li>
<strong>Poor library preparation</strong>: can result in RNAseq
reads that consist primarily of adapter sequences without any actual
biological content - this can result in too many unmapped reads.</li>
<li>
<strong>Over-processing</strong> Short reads from over-trimming can
shorten reads to the point where they are too short to align uniquely,
leading to either multimapping or unmapped reads.</li>
</ul><p>It is crucial to check the quality of the reads before mapping and
also alignment statistics after mapping.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="preparing-for-read-mapping">Preparing for read mapping<a class="anchor" aria-label="anchor" href="#preparing-for-read-mapping"></a></h2>
<hr class="half-width"><p>Before we start mapping the reads, we need to prepare the reference
genome. Mapping programs require the reference genome to be indexed.
Indexing is the process of creating a data structure that allows the
mapping program to quickly search the reference genome for the best
alignment of the reads. <code>STAR</code> requires the reference genome
to be indexed before mapping. Mapping programs can also use
<code>GTF</code> or <code>GFF</code> files to guide the alignment
process. These files contain information about the gene models in the
reference genome, such as the location of exons and introns, that can
speed up the alignment process.</p>
<p>We will use the reference genome and annotations that we downloaded
in the previous episode and index it using <code>STAR</code>.</p>
</section><section><h2 class="section-heading" id="indexing">Indexing<a class="anchor" aria-label="anchor" href="#indexing"></a></h2>
<hr class="half-width"><p>STAR has <code>--runMode</code> option where you can alter the
function of the program. When <code>genomeGenerate</code> is specified
with this option, it can be used for indexing.</p>
<div id="generating-genome-indexes" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="generating-genome-indexes" class="callout-inner">
<h3 class="callout-title">Generating genome indexes</h3>
<div class="callout-content">
<p>There are many options for the <code>STAR</code> program to index the
reference genome. You will need the reference genome fasta file that
needs to indexed, but there are many other options as well. Can you <a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf" class="external-link">check
the manual</a> and find options that are important for indexing the
reference genome?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>In page 5, Section 2, “Generating genome indexes” discuss the options
for indexing the reference genome. You will need to use
<code>--runMode genomeGenerate</code> for indexing. Important options
for indexing are:</p>
<ul><li>
<code>runThreadN</code> number of threads to use</li>
<li>
<code>genomeDir</code> output directory for the genome indexes</li>
<li>
<code>genomeFastaFiles</code> reference genome fasta file</li>
<li>
<code>sjdbGTFfile</code> the GTF file</li>
<li>
<code>sjdbOverhang</code> default value of 100 works well, for most
cases this can be omitted</li>
</ul></div>
</div>
</div>
</div>
<p>We will also need SLURM directives to submit the job to the cluster.
The number of threads to use can be specified using the
<code>--cpus-per-task</code> directive. The script to index the
reference genome using <code>STAR</code> would look like this:</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=1</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#SBATCH --cpus-per-task=4</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#SBATCH --time=1:00:00</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co">#SBATCH --job-name=index_genome</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">#SBATCH --output=cluster-%x.%j.out</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#SBATCH --error=cluster-%x.%j.out</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="ex">ml</span> star</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="va">genome</span><span class="op">=</span><span class="st">"/home/aseethar/rcac_rnaseq/data/GRCm39.primary_assembly.genome.fa"</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="va">genomeDir</span><span class="op">=</span><span class="st">"/home/aseethar/rcac_rnaseq/data/star_index"</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="va">gtf</span><span class="op">=</span><span class="st">"/home/aseethar/rcac_rnaseq/data/gencode.vM35.primary_assembly.basic.annotation.gtf"</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="ex">STAR</span> <span class="dt">\</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>   <span class="at">--runThreadN</span> 4<span class="dt">\</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>   <span class="at">--runMode</span> genomeGenerate <span class="dt">\</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>   <span class="at">--genomeDir</span> <span class="va">${genomeDir}</span> <span class="dt">\</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>   <span class="at">--genomeFastaFiles</span> <span class="va">${genome}</span> <span class="dt">\</span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>   <span class="at">--sjdbGTFfile</span> <span class="va">${gtf}</span></span></code></pre>
</div>
<p>The script can be saved in a file, say <code>index_genome.sh</code>,
within the <code>scripts</code> directory. It should be submitted as a
slurm job after making it executable.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">chmod</span> +x index_genome.sh</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">sbatch</span> index_genome.sh</span></code></pre>
</div>
<p>It takes about 30-45 minutes to complete the indexing process.</p>
<div id="is-gtf-file-necessary-for-indexing" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="is-gtf-file-necessary-for-indexing" class="callout-inner">
<h3 class="callout-title">Is GTF file necessary for indexing?</h3>
<div class="callout-content">
<p>The GTF file is not necessary for indexing the reference genome, but
will be useful for alignment process. The GTF file contains information
about the gene models in the reference genome, such as the location of
exons and introns, that can speed up the alignment process. If you have
a GTF file, it is recommended to use it when indexing the reference
genome.</p>
</div>
</div>
</div>
<div id="can-i-use-gff3-file-instead" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="can-i-use-gff3-file-instead" class="callout-inner">
<h3 class="callout-title">Can I use GFF3 file instead?</h3>
<div class="callout-content">
<p>Yes, you can use GFF3 file instead of GTF file. STAR can handle both
GTF and GFF3 files. In addition to the aforementioned options, for GFF3
formatted annotations you need to use
<code>--sjdbGTFtagExonParentTranscript Parent</code>. But if you have
issues with GFF3, you can convert it to GTF using <code>gffread</code>
from the <code>Cufflinks</code> package.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="mapping-reads-to-the-reference-genome">Mapping reads to the reference genome<a class="anchor" aria-label="anchor" href="#mapping-reads-to-the-reference-genome"></a></h2>
<hr class="half-width"><p>After indexing, our genome is ready for mapping reads. The
<code>STAR</code> program has many options that can be used to fine-tune
the mapping process. You can <a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf" class="external-link">check
the manual</a> and find options that are important for mapping reads to
the reference genome.</p>
<p>For mapping, you will need to use <code>--runMode alignReads</code>
important options for aligning reads are:</p>
<ul><li>
<code>runThreadN</code> number of threads to use</li>
<li>
<code>readFilesIn</code> fastq files to be aligned, can be single (1
file) or paired-end (2 files)</li>
<li>
<code>genomeDir</code> reference genome index directory</li>
</ul><p>For output options, you can use:</p>
<ul><li>
<code>outFileNamePrefix</code> prefix for all output files</li>
<li>
<code>outSAMtype</code>: output filetype (SAM default), but can be
BAM, CRAM, etc.</li>
<li>
<code>outSAMunmapped</code>: handling of unmapped reads</li>
</ul><div id="what-other-mapping-options-should-i-consider" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="what-other-mapping-options-should-i-consider" class="callout-inner">
<h3 class="callout-title">What other mapping options should I consider?</h3>
<div class="callout-content">
<p>STAR’s default parameters are optimized for mammalian genomes. If you
are using for other genomes like plant, fungi or any other species, you
may need to fine-tune your mapping options:</p>
<ul><li>When using compressed reads, option <code>readFilesCommand</code>
with <em>UncompressionCommand</em> can be used. If using gzipped files
(<code>*.gz</code>) use <code>--readFilesCommand zcat</code> OR
<code>--readFilesCommand gunzip -c</code>. For <code>bzip2</code>
compressed files, use <code>--readFilesCommand bunzip2 -c</code>.</li>
<li>For controling the multi-mapped reads, options like
<code>outFilterMultimapNmax</code> (default 10) and
<code>winAnchorMultimapNmax</code> can be used. Useful when you have
polyploid genomes or repetitive regions.</li>
<li>For controlling mapping sensitivity, options like
<code>outFilterMismatchNmax</code> (default 10) and
<code>outFilterMismatchNoverLmax</code> can be used.</li>
<li>Shorter reads may require lower <code>outFilterMismatchNmax</code>
and <code>outFilterMismatchNoverLmax</code> values.</li>
</ul></div>
</div>
</div>
<div id="script-for-mapping-reads" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="script-for-mapping-reads" class="callout-inner">
<h3 class="callout-title">Script for mapping reads</h3>
<div class="callout-content">
<p>Write a script to map the reads to the reference genome using
<code>STAR</code>. You will need the reference genome index directory,
fastq files to be aligned, and output directory. Can you write a mapping
command for <code>STAR</code>?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">STAR</span> <span class="at">--runMode</span> alignReads <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>     <span class="at">--runThreadN</span> 12 <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>     <span class="at">--genomeDir</span> /home/aseethar/rcac_rnaseq/data/star_index <span class="dt">\</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>     <span class="at">--readFilesIn</span> SRR1234567_1.fastq.gz SRR1234567_2.fastq <span class="dt">\</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>     <span class="at">--outFileNamePrefix</span> SRR1234567 <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>     <span class="at">--outSAMtype</span> BAM SortedByCoordinate <span class="dt">\</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>     <span class="at">--outSAMunmapped</span> Within</span></code></pre>
</div>
</div>
</div>
</div>
</div>
<p>Since we have 45 sets of fastq files, we can write an array job for
mapping all the reads. The script to map the reads using
<code>STAR</code> would look like this:</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#SBATCH --ntasks=1</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#SBATCH --cpus-per-task=12</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#SBATCH --time=8:00:00</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#SBATCH --job-name=read_mapping</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#SBATCH --array=1-45    </span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#SBATCH --output=cluster-%x.%j.out</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#SBATCH --error=cluster-%x.%j.out</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="ex">ml</span> biocontainers</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="ex">ml</span> star</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co"># Define paths</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="va">FASTQ_DIR</span><span class="op">=</span><span class="st">"/home/aseethar/rcac_rnaseq/data/fastq"</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="va">GENOME_INDEX</span><span class="op">=</span><span class="st">"/home/aseethar/rcac_rnaseq/data/star_index"</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="va">OUTPUT_DIR</span><span class="op">=</span><span class="st">"/home/aseethar/rcac_rnaseq/data/mapping"</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co"># Create an array of FASTQ file pairs</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="va">FASTQ_R1</span><span class="op">=</span><span class="va">($FASTQ_DIR</span>/<span class="pp">*</span>_R1_<span class="pp">*</span>.fastq.gz<span class="va">)</span>   <span class="co"># List of all R1 files</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="va">FASTQ_R2</span><span class="op">=</span><span class="va">($FASTQ_DIR</span>/<span class="pp">*</span>_R2_<span class="pp">*</span>.fastq.gz<span class="va">)</span>   <span class="co"># List of all R2 files</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co"># Get the FASTQ file for this array job</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="va">R1</span><span class="op">=</span><span class="va">${FASTQ_R1</span><span class="op">[</span><span class="va">$SLURM_ARRAY_TASK_ID</span>-1<span class="op">]</span><span class="va">}</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="va">R2</span><span class="op">=</span><span class="va">${FASTQ_R2</span><span class="op">[</span><span class="va">$SLURM_ARRAY_TASK_ID</span>-1<span class="op">]</span><span class="va">}</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co"># Extract sample name from FASTQ file</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="va">SAMPLE</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$R1</span> _R1_001.fastq.gz<span class="va">)</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co"># Run STAR alignment for the current pair of FASTQ files</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a><span class="ex">STAR</span> <span class="at">--runThreadN</span> 4 <span class="dt">\</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>     <span class="at">--genomeDir</span> <span class="va">$GENOME_INDEX</span> <span class="dt">\</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>     <span class="at">--readFilesIn</span> <span class="va">$R1</span> <span class="va">$R2</span> <span class="dt">\</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>     <span class="at">--readFilesCommand</span> zcat <span class="dt">\</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>     <span class="at">--outFileNamePrefix</span> <span class="va">$OUTPUT_DIR</span>/<span class="va">$SAMPLE</span> <span class="dt">\</span></span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>     <span class="at">--outSAMunmapped</span> Within <span class="dt">\</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>     <span class="at">--outSAMtype</span> BAM SortedByCoordinate</span></code></pre>
</div>
<p>The job can be submitted as an array job using the following
command:</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">chmod</span> +x map_reads.sh</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">sbatch</span> <span class="at">--array</span><span class="op">=</span>1-45 map_reads.sh</span></code></pre>
</div>
<p>Since we have already finished mapping the reads, we can skip this
and move on to the next step.</p>
<div id="can-i-use-star-for-counting-reads" class="callout prereq">
<div class="callout-square">
<i class="callout-icon" data-feather="check"></i>
</div>
<div id="can-i-use-star-for-counting-reads" class="callout-inner">
<h3 class="callout-title">Can I use STAR for counting reads?</h3>
<div class="callout-content">
<p>Yes, STAR can be used for counting reads. STAR can output the number
of reads that map to each gene in the reference genome. This can be
useful for differential gene expression analysis. You can use the
<code>--quantMode</code> option to enable read counting. The
<code>--quantMode</code> option can be set to <code>GeneCounts</code> to
output the number of reads that map to each gene.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul><li><p>Mapping Raw Reads: The process of mapping involves aligning short
reads from sequencing data to a reference genome to determine their
locations. STAR, a splice-aware RNA-seq aligner, is commonly used due to
its speed and accuracy in handling exon-exon junctions.</p></li>
<li><p>Impact of Read Quality: Poor read quality, such as
over-represented sequences or low-quality bases, can lead to unmapped or
multi-mapped reads. It is important to check the quality of reads before
mapping and review alignment statistics after mapping to ensure data
quality.</p></li>
<li><p>Reference Genome Indexing: Before mapping, the reference genome
needs to be indexed using tools like STAR. Indexing creates a data
structure to facilitate rapid alignment, and additional files like
GTF/GFF annotations can improve alignment accuracy.</p></li>
<li><p>Optimizing Mapping: Fine-tuning mapping parameters (e.g.,
handling multi-mapped reads, controlling sensitivity) based on the
dataset is essential. The STAR aligner offers numerous options that
allow adjustment for specific genome types and read qualities.</p></li>
</ul></div>
</div>
</div>
<!--
Place links that you need to refer to multiple times across pages here. Delete
any links that you are not going to use.
 -->
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="01a-from-reads-to-counts.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="01c-count-features.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="01a-from-reads-to-counts.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Downloading and
        </a>
        <a class="chapter-link float-end" href="01c-count-features.html" rel="next">
          Next: Quantifying Gene...
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://aseetharam.github.io/rcac_rnaseq_workshop/edit/main/episodes/01b-mapping-reads.Rmd" class="external-link">Edit on GitHub</a>

	
        | <a href="https://aseetharam.github.io/rcac_rnaseq_workshop/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://aseetharam.github.io/rcac_rnaseq_workshop/" class="external-link">Source</a></p>
				<p><a href="https://aseetharam.github.io/rcac_rnaseq_workshop/blob/main/CITATION.cff" class="external-link">Cite</a> | <a href="mailto:aseethar@purdue.edu">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.7" class="external-link">sandpaper (0.16.7)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.6" class="external-link">pegboard (0.7.6)</a>, and <a href="https://github.com/aseetharam/varnish/tree/4189f1f22ae1372d077a4d4c9a01d27441264c56" class="external-link">varnish (1.0.4)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://rcac_rnaseq_workshop.github.io/NA/01b-mapping-reads.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, RNA-seq, Bioconductor",
  "name": "Mapping reads to a reference genome",
  "creativeWorkStatus": "active",
  "url": "https://rcac_rnaseq_workshop.github.io/NA/01b-mapping-reads.html",
  "identifier": "https://rcac_rnaseq_workshop.github.io/NA/01b-mapping-reads.html",
  "dateCreated": "2024-08-30",
  "dateModified": "2024-09-10",
  "datePublished": "2024-09-10"
}

  </script><script>
		feather.replace();
	</script></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

